app/controllers/publishers/vacancies/job_applications/messages_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Publishers::Vacancies::JobApplications::MessagesController < Publishers::Vacancies::JobApplications::BaseController
+  before_action :set_job_application
+
+  def create
+    conversation = find_or_create_conversation
+    message = conversation.messages.build(message_form_params.merge(sender: current_publisher))
+
+    if message.valid?
+      conversation.save! unless conversation.persisted?
+      message.save!
+      redirect_to organisation_job_job_application_path(id: @job_application.id, tab: "messages"), success: t(".success")
+    else
+      redirect_to organisation_job_job_application_path(id: @job_application.id, tab: "messages"), warning: t(".failure")
+    end
+  end
+
+  private
+
+  def find_or_create_conversation
+    @job_application.conversations.first ||
```

Comment:
I'm a little unsure about this code. Looks like we can only have 1 conversation?

Reply:
Yeah, I just took this ticket as an MVP of the messaging functionality. Sounds like we will be able to have multiple conversations per job application at some point but I don't really know what that will look like based on the tickets on the trello board so for now I was just working on the assumption that the MVP will only have one conversation.